benchexec/tools/ultimate.py

```diff
@@ -204,113 +186,120 @@ def _requires_ultimate_data(self, executable):
+
+        if mem_bytes:
+            cmdline += ["-Xmx" + str(mem_bytes)]
+        cmdline += ["-Xss4m"]
+        cmdline += ["-jar", self._get_current_launcher_jar(executable)]
+
+        if self._requires_ultimate_data(executable):
+            if "-ultimatedata" not in options and "-data" not in options:
+                if self.api == 2:
+                    cmdline += [
+                        "-data",
+                        "@noDefault",
+                        "-ultimatedata",
+                        os.path.join(os.path.dirname(executable), "data"),
+                    ]
+                if self.api == 1:
+                    raise ValueError(
+                        "Illegal option -ultimatedata for API {} and Ultimate version {}".format(
+                            self.api, self.version(executable)
                         )
-                elif "-ultimatedata" in options and "-data" not in options:
-                    if self.api == 2:
-                        cmdline += ["-data", "@noDefault"]
-                    if self.api == 1:
-                        raise ValueError(
-                            "Illegal option -ultimatedata for API {} and Ultimate version {}".format(
-                                self.api, self.version(executable)
-                            )
+                    )
+            elif "-ultimatedata" in options and "-data" not in options:
+                if self.api == 2:
+                    cmdline += ["-data", "@noDefault"]
+                if self.api == 1:
+                    raise ValueError(
+                        "Illegal option -ultimatedata for API {} and Ultimate version {}".format(
+                            self.api, self.version(executable)
                         )
-            else:
-                if "-data" not in options:
-                    if self.api == 2 or self.api == 1:
-                        cmdline += [
-                            "-data",
-                            os.path.join(os.path.dirname(executable), "data"),
-                        ]
-
-            cmdline += options
-
-            if tasks:
-                cmdline += ["-i"] + tasks
-            self.__assert_cmdline(
-                cmdline,
-                "cmdline contains empty or None argument when using Ultimate raw mode: ",
-            )
-            return cmdline
-
-        # there is no way to run ultimate; not enough parameters
-        raise UnsupportedFeatureException(
-            "Unsupported argument combination: options={} propertyfile={} rlimits={}".format(
-                options, propertyfile, rlimits
-            )
+                    )
+        else:
+            if "-data" not in options:
+                if self.api == 2 or self.api == 1:
+                    cmdline += [
+                        "-data",
+                        os.path.join(os.path.dirname(executable), "data"),
+                    ]
+
+        cmdline += options
+
+        if task.input_files:
```

Comment:
`task.input_files` will always be non-empty when it is returned, otherwise an exception is thrown. If you want to support both empty and non-empty list of input files, you would need to use
```suggestion
        if task.input_files_or_empty:
```